### PR TITLE
Adds replication and adoption for module import

### LINKED ISSUE #9361: Add replication and association booleans to Module bulk import
### NetBox version

v3.2.3

### Feature type

Change to existing functionality

### Proposed functionality

It would be nice to extend the replication and association checkboxes available in the UI to also be available when doing a module bulk import.

### Use case

Adding a single module at a time is tedious and we are looking to utilize more bulk imports, but have now run into the limitation that device components cannot be automatically associated with that method and would still require manual intervention after the fact to do those associations. Possibly the larger at this point though is that there is no way to disable the component replication, so trying to do an import like this generates an exception for duplicate keys.

### Database changes

None

### External dependencies

None

### PR DESCRIPTION
**Pull request by sleepinggenius2**
### Fixes: #9361

Makes the replicate_components and adopt_components fields available in the UI to also be available when doing a module bulk import.

### CODE REVIEW

**Comment by kkthxbye-code** [comment]
Thank you for the PR! I'm not a big fan of having that much duplicate code for the clean method.

@jeremystretch is there any clean way to share the `clean` method between the model form and the CSV form? I guess we could create a mixin with a `clean_module` method, just not sure where to put it.

The code functioned as expected when testing.

**Comment by jeremystretch** [comment]
I don't have a strong preference for an approach, but we definitely can't just replicate the existing code. Maybe a utility function within the `dcim.forms` module?

**Comment by sleepinggenius2** [comment]
Updated PR to move clean logic to a common class that both forms can inherit from. This works in my testing, but I'm not very familiar with all of this, so please let me know if this isn't a good solution.

**Comment by sleepinggenius2** [comment]
As we're getting closer to the v3.3 release, I just wanted to follow up on this to see if there's anything else I need to do to be able to get this merged.

**Comment by sleepinggenius2** [comment]
@kkthxbye-code or @jeremystretch, as there have been a number of point releases now since this PR was submitted and we continue to get closer to a v3.3 release, I just wanted to follow up again to see if there is anything I still need to do to get this merged. Thank you.

**Review by jeremystretch — DISMISSED** [review]
@sleepinggenius2 I think the approach looks good, thanks for the revision. Could you extend the import view test to verify the function of the two new fields please?

**Comment by sleepinggenius2** [comment]
> @sleepinggenius2 I think the approach looks good, thanks for the revision. Could you extend the import view test to verify the function of the two new fields please?

I assume that this will need to go in the `ModuleTestCase` class of `netbox/dcim/tests/test_views.py`? I'm not particularly familiar with testing in Python, but I think I have enough information to give it a shot.

**Comment by sleepinggenius2** [comment]
@jeremystretch, I requested a review a month ago after I made the changes that you asked for, but it is still showing *Changes requested* and there has been no movement, so I apologize if there was something I missed in that process.

**Comment by sleepinggenius2** [comment]
@jeremystretch or @kkthxbye-code, what do I need to do to get this merged or at least kept from being closed?

**Review by kkthxbye-code — APPROVED** [review]
Sorry this fell through the cracks @sleepinggenius2

I checked out the PR and rebased on the development and everything works exactly as it should, both for CSV import and through the UI for all combinations of replication/adoption. So this looks good to me, can we merge this @jeremystretch ?

File: netbox/dcim/tests/test_views.py [hunk 1]
```diff
@@ -1907,6 +1954,49 @@ def test_module_component_adoption(self):
...
+        device = Device.objects.get(pk=form_data['device'])
+
+        # Create an interface to be adopted
+        interface = Interface(device=device, name=interface_name, type=InterfaceTypeChoices.TYPE_10GE_FIXED)
+        interface.save()
+
+        # Ensure that interface is created with no module
+        self.assertIsNone(interface.module)
+
+        # Create a module with adopted components
```
Comment (kkthxbye-code) [inline netbox/dcim/tests/test_views.py:1977]: One small thing, this testcase actually fails on my PC. It's trying to install in the first modulebay of the device, which is already occupied from the setup. If I change to another bay it passes. Not sure if it's something with my setup or from the rebase.

Comment (jeremystretch) [reply netbox/dcim/tests/test_views.py:1977]: Same. Maybe related to 9ef24d3f43022f61044756741f40e095e9290bc1? In any case, changing it to an empty bay appears to fix the test.

Comment (jeremystretch) [reply netbox/dcim/tests/test_views.py:1955]: Pushed a fix for this in [2a55c29](https://github.com/netbox-community/netbox/pull/9498/commits/2a55c29a9a09fe20a61dcc9964e298769ada3e72).

**Comment by sleepinggenius2** [comment]
I appreciate your all's assistance with this.

**Comment by jeremystretch** [comment]
Not sure what's going on with CI today. Tests pass locally, though I'm getting failure notifications from GitHub. Nothing's showing up for the PR currently.

Going to go ahead and merge this into `develop` and deal with any remaining issues there. Thanks for your work on this @sleepinggenius2 and sorry for the delays!

**Pull request merged**